lib/libzpool/uzfs_mtree.c

```diff
@@ -302,3 +304,152 @@ uzfs_destroy_txg_diff_tree(void *tree)
 	avl_destroy(temp_tree);
 	umem_free(temp_tree, sizeof (*temp_tree));
 }
+
+void
+uzfs_add_to_incoming_io_tree(zvol_state_t *zv, uint64_t offset, uint64_t len)
+{
+	/*
+	 * Here: Handling of incoming_io_tree creation is for error case only.
+	 *	 It should be handled by replica or caller of uzfs_write_data
+	 */
+	if (!zv->incoming_io_tree)
```

Comment:
incoming_io_tree should be protected with lock

Reply:
ok